addons/marketing_card/controllers/marketing_card.py

```diff
@@ -0,0 +1,129 @@
+import base64
+
+from odoo.http import Controller, content_disposition, request, route
+from odoo.tools import consteq
+
+from .user_agents import NETWORK_TO_AGENT
+from ..utils.image_utils import scale_image
+
+class MarketingCardController(Controller):
+
+    @route(['/cards/<int:campaign_id>/<int:res_id>/<string:hash_token>/card.jpg'], type='http', auth='public', sitemap=False, website=True)
+    def card_campaign_image(self, campaign_id, res_id, hash_token, small=False):
+        campaign_sudo = request.env['card.campaign'].sudo().browse(campaign_id).exists()
+        if not campaign_sudo or not consteq(hash_token, campaign_sudo._generate_card_hash_token(res_id)):
+            raise request.not_found()
+
+        target_sudo = request.env[campaign_sudo.res_model].sudo().browse(res_id).exists()
+        if not target_sudo:
+            return request.not_found()
+
+        card_sudo = campaign_sudo._get_or_create_card_from_res_id(res_id)
+
+        crawler = self._get_crawler_name(request)
+        if crawler and not card_sudo.is_shared:
+            request.env['bus.bus']._sendone(f'card_shared_target-{campaign_id}-{hash_token}', 'marketing_card/share_card_target', {
+                'message': campaign_sudo.thanks_message,
+                'reward_url': campaign_sudo.thanks_redirection,
+            })
+            card_sudo.is_shared = True
+
+        if all(element.value_type == 'static' for element in campaign_sudo.element_ids):
+            image_bytes = base64.b64decode(campaign_sudo.image_preview)
+        else:
+            image_bytes = base64.b64decode(card_sudo._get_or_generate_image())
+        image_bytes = image_bytes if not small else scale_image(image_bytes, 0.5)
+        return request.make_response(image_bytes, [
+            ('Content-Type', ' image/jpeg'),
+            ('Content-Length', len(image_bytes)),
+            ('Content-Disposition', content_disposition('card.jpg')),
+        ])
+
+    @route(['/cards/<int:campaign_id>/card.jpg'])
+    def card_campaign_image_placeholder(self, campaign_id):
```

Comment:
The preview from the form will not necessarily be a "generic" image anymore, so this is now needed (indeed it may not have been when this comment was written)